lib/market.rb

```diff
@@ -0,0 +1,69 @@
+class Market 
+  attr_reader :name, :vendors, :add_vendor, :total_inventory
+  def initialize(name)
+    @name = name
+    @vendors = []
+  end
+
+  def add_vendor(vendor)
+    @vendors << vendor
+
+  end
+
+  def vendor_names
+    @vendors.map {|vendor| vendor.name}
+    #array of vendor names 
+  end
+
+  def vendors_that_sell(item_name)
+    v_that_s = []
+    @vendors.each do |vendor|
+      if vendor.check_stock(item_name) != 0
+        v_that_s << vendor.name
```

Comment:
This should return the vendor object, not just their name
```suggestion
        v_that_s << vendor
```

Reply:
This method can also be cleaned up quite a bit using a `select` or `find_all` instead of using `each` with a collector array.